Python/ast_unparse.c

```diff
@@ -58,6 +59,24 @@ append_charp(_PyUnicodeWriter *writer, const char *charp)
         } \
     } while (0)
 
+static PyObject *
+replace_inf_value(PyObject *repr)
+{
+    PyObject *inf_str = PyUnicode_InternFromString("inf");
+    if (!inf_str) {
+        return NULL;
+    }
+    PyObject *inf_value = PyUnicode_FromFormat("1e%d", 1 + DBL_MAX_10_EXP);
+    if (!inf_value) {
+        return NULL;
+    }
+    PyObject *result = PyUnicode_Replace(repr, inf_str, inf_value, -1);
+    Py_DECREF(repr);
```

Comment:
I dislike functions modifying the input argument. I would prefer to leave repr argument unchanged. Call Py_DECREF(repr) in the caller site.

Reply:
If I correctly understand you that might look bad because `replace_inf_value` doesn't return `repr`, it returns another string (`result`) which might be same or not. In any case if I move that decreasing refcount call to caller site, I need to keep a individual reference to inital `repr` object for decrefing after assignment.

Reply:
>  if I move that decreasing refcount call to caller site, I need to keep a individual reference to inital repr object for decrefing after assignment.

Exactly. I would prefer such explicit code.